File: common/chatRoom.ts
import { ChatRoomMessage } from './chatRoomMessage';
import { User } from './user';
export class ChatRoom {
    chatRoomName: string;
    users: User[] = [];
    currentMessage: ChatRoomMessage;
    messages: ChatRoomMessage[];
    creator: User;
    chatRoomId: string;

    constructor(roomId: string, creator: User, chatRoomName: string) {
        this.chatRoomId = roomId;
        this.users.push(creator);
        this.currentMessage = new ChatRoomMessage('', '', '');
        this.messages = [];
        this.creator = creator;
        this.chatRoomName = chatRoomName;
    }
}

File: prototypeLeger/app/src/main/java/com/example/scrabbleprototype/model/ChatRoom.kt
package com.example.scrabbleprototype.model

import kotlinx.serialization.Serializable

@Serializable
class ChatRoom (var chatRoomId: String, var creator: User, var chatRoomName: String) {
    var users: ArrayList<User> = arrayListOf(creator)
    var currentMessage: ChatRoomMessage = ChatRoomMessage("", "", "")
    var messages: ArrayList<ChatRoomMessage> = arrayListOf()

    fun containsUser(user: User): Boolean {
        return users.any { it.pseudonym == user.pseudonym }
    }
}
